nodes/src/nodes/rerank_cohere/IInstance.py

```
@@ -0,0 +1,120 @@
+
+        The question's documents are reranked using the Cohere Rerank API,
+        then filtered by the configured min_score threshold. The reranked
+        documents are written to the documents output lane, and an answer
+        with the reranked documents is written to the answers output lane.
+        """
+        # Deep copy to avoid mutating the shared question object in fan-out pipelines
+        question = copy.deepcopy(question)
+
+        reranker = self.IGlobal._reranker
+        if reranker is None:
+            raise Exception('Reranker not initialized')
```

Ruff (0.15.7)

[warning] Create your own exception

(TRY002)